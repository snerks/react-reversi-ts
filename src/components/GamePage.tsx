import * as React from 'react';

export interface GamePageProps {
}

class GamePage extends React.Component<GamePageProps, {}> {
    render() {
        return (
            <div>
                GamePage
            </div>
        );
    }
}

export default GamePage;
