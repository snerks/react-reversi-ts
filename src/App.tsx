import * as React from 'react';

import GamePage from './components/GamePage';

class App extends React.Component<null, null> {
  render() {
    return (
      <div className="container-fluid">
        <nav className="navbar navbar-inverse">
          <div className="container-fluid">
            <div className="navbar-header">
              <button
                type="button"
                className="navbar-toggle collapsed"
                data-toggle="collapse"
                data-target="#navbar"
                aria-expanded="false"
                aria-controls="navbar"
              >
                <span className="sr-only">Toggle navigation</span>
                <span className="icon-bar" />
                <span className="icon-bar" />
                <span className="icon-bar" />
              </button>
              <a className="navbar-brand" href="#">Othello TS React Redux</a>
            </div>
            <div id="navbar" className="navbar-collapse collapse">
              <ul className="nav navbar-nav">
                <li className="active"><a href="#">Home</a></li>
                <li><a href="https://github.com/snerks/react-reversi-ts">Code</a></li>
              </ul>
            </div>
          </div>
        </nav>
        <div className="text-center">
          <GamePage />
        </div>
      </div>
    );
  }
}

export default App;
