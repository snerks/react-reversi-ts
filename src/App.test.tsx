import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import App from './App';
import GamePage from './components/GamePage';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <App />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  const _component = shallow(
    <App />
  );

  it('contains a GamePage subcomponent', () => {
    expect(_component.find(GamePage)).toHaveLength(1);
  });
});
