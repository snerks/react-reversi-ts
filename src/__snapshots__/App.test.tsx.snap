exports[`App renders and matches our snapshot 1`] = `
<div
  className="container-fluid">
  <nav
    className="navbar navbar-inverse">
    <div
      className="container-fluid">
      <div
        className="navbar-header">
        <button
          aria-controls="navbar"
          aria-expanded="false"
          className="navbar-toggle collapsed"
          data-target="#navbar"
          data-toggle="collapse"
          type="button">
          <span
            className="sr-only">
            Toggle navigation
          </span>
          <span
            className="icon-bar" />
          <span
            className="icon-bar" />
          <span
            className="icon-bar" />
        </button>
        <a
          className="navbar-brand"
          href="#">
          Othello TS React Redux
        </a>
      </div>
      <div
        className="navbar-collapse collapse"
        id="navbar">
        <ul
          className="nav navbar-nav">
          <li
            className="active">
            <a
              href="#">
              Home
            </a>
          </li>
          <li>
            <a
              href="https://github.com/snerks/react-reversi-ts">
              Code
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="text-center">
    <div>
      GamePage
    </div>
  </div>
</div>
`;
